src/core/server/saved_objects/migrationsv2/migrations_state_action_machine.ts

```diff
@@ -195,8 +140,11 @@ export async function migrationStateActionMachine({
         };
       }
     } else if (finalState.controlState === 'FATAL') {
-      await cleanup(client, executionLog, finalState);
-      dumpExecutionLog(logger, logMessagePrefix, executionLog);
```

Comment:
Looking at the code, it seems to be the case, but just to be sure:

1. When the migration logger's level is `DEBUG`, we're still outputting the same information we previously stored in the executionLogs, right?

2. In case of failure, the full cause is still outputted in the logs even when not in `DEBUG` level?

Reply:
1. Yes
2. We still output the full fatal message + traces etc. The difference is, before we would also log the execution log including the full state and responses to each action. Because this was a `meta` field, it wasn't visible by default with the legacy logger config, you would have to switch to NP JSON logging to actually see this. So in practice for most on-prem and cloud users we won't loose any details.

Reply:
I missed that we only logged the state transition meta information from the executionLog. So I'm adding debug logging when there's a state transition so that we can inspect the complete state after every step when debugging is enabled.